core/capabilities/ccip/ccipsolana/executecodec.go

```diff
@@ -0,0 +1,263 @@
+package ccipsolana
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	agbinary "github.com/gagliardetto/binary"
+	"github.com/gagliardetto/solana-go"
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_router"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// ExecutePluginCodecV1 is a codec for encoding and decoding execute plugin reports.
+// Compatible with:
+// - "OffRamp 1.6.0-dev"
+type ExecutePluginCodecV1 struct {
+}
+
+func NewExecutePluginCodecV1() *ExecutePluginCodecV1 {
+	return &ExecutePluginCodecV1{}
+}
+
+func (e *ExecutePluginCodecV1) Encode(ctx context.Context, report cciptypes.ExecutePluginReport) ([]byte, error) {
+	if len(report.ChainReports) != 1 {
+		return nil, fmt.Errorf("unexpected chain report length: %d", len(report.ChainReports))
+	}
+
+	chainReport := report.ChainReports[0]
+	if len(chainReport.Messages) > 1 {
+		return nil, fmt.Errorf("unexpected report message length: %d", len(chainReport.Messages))
+	}
+
+	var message ccip_router.Any2SolanaRampMessage
+	var offChainTokenData [][]byte
+	if len(chainReport.Messages) > 0 {
+		// currently only allow executing one message at a time
+		msg := chainReport.Messages[0]
+		tokenAmounts := make([]ccip_router.Any2SolanaTokenTransfer, 0, len(msg.TokenAmounts))
+		for _, tokenAmount := range msg.TokenAmounts {
+			if tokenAmount.Amount.IsEmpty() {
+				return nil, fmt.Errorf("empty amount for token: %s", tokenAmount.DestTokenAddress)
+			}
+
+			if len(tokenAmount.DestTokenAddress) != solana.PublicKeyLength {
+				return nil, fmt.Errorf("invalid destTokenAddress address: %v", tokenAmount.DestTokenAddress)
+			}
+
+			tokenAmounts = append(tokenAmounts, ccip_router.Any2SolanaTokenTransfer{
+				SourcePoolAddress: tokenAmount.SourcePoolAddress,
+				DestTokenAddress:  solana.PublicKeyFromBytes(tokenAmount.DestTokenAddress),
+				ExtraData:         tokenAmount.ExtraData,
+				Amount:            bigIntToBytes32(tokenAmount.Amount),
```

Comment:
As per latest solana go code, this should be LE:
https://github.com/smartcontractkit/chainlink-ccip/blob/a0491362f884640d8734bd1d5eae5d25a394a5ce/chains/solana/gobindings/ccip_router/types.go#L943-L945

https://github.com/smartcontractkit/chainlink-ccip/blob/a0491362f884640d8734bd1d5eae5d25a394a5ce/chains/solana/gobindings/ccip_router/types.go#L708

Also, you might need to refresh and use Any2SVMTokenTransfer instead of Any2SolanaTokenTransfer